src/model.c

```diff
@@ -14,17 +14,130 @@
 
 RedisModuleType *RedisAI_ModelType = NULL;
 
-static void* Model_RdbLoad(struct RedisModuleIO *io, int encver) {
-  //todo
-  return NULL;
+static void* RAI_Model_RdbLoad(struct RedisModuleIO *io, int encver) {
+  if (encver != RAI_ENC_VER) {
+      /* We should actually log an error here, or try to implement
+         the ability to load older versions of our data structure. */
+      return NULL;
+  }
+
+  RAI_Backend backend = RedisModule_LoadUnsigned(io);
+  RAI_Device device = RedisModule_LoadUnsigned(io);
+  size_t ninputs = RedisModule_LoadUnsigned(io);
+
+  const char **inputs = RedisModule_Alloc(ninputs * sizeof(char*));
+
+  for (size_t i=0; i<ninputs; i++) {
+    inputs[i] = RedisModule_LoadStringBuffer(io, NULL);
+  }
+
+  size_t noutputs = RedisModule_LoadUnsigned(io);
+
+  const char **outputs = RedisModule_Alloc(ninputs * sizeof(char*));
+
+  for (size_t i=0; i<noutputs; i++) {
+    outputs[i] = RedisModule_LoadStringBuffer(io, NULL);
+  }
+
+  size_t len;
+
+  char *buffer = RedisModule_LoadStringBuffer(io, &len);
+
+  RAI_Error err = RAI_InitError();
+
+  RAI_Model *model = RAI_ModelCreate(backend, device, ninputs, inputs, noutputs, outputs,
+                                     buffer, len, &err);
+ 
+  // TODO: ERR
+
+  RedisModule_Free(inputs);
+  RedisModule_Free(outputs);
+  RedisModule_Free(buffer);
+
+  return model;
 }
 
-static void Model_RdbSave(RedisModuleIO *rdb, void *value) {
-  //todo
+static void RAI_Model_RdbSave(RedisModuleIO *io, void *value) {
+  RAI_Model *model = (RAI_Model*)value;
+  char *buffer = NULL;
+  size_t len = 0;
+  RAI_Error err = RAI_InitError();
```

Comment:
I should've commented on this earlier, but zero-initialization is preferred over structure by-value returns.

Reply:
Done using zero-initialization